### PR TITLE
Publish docker image as part of the release process

### LINKED ISSUE #798: Publish docker images as part of the release process

### PR DESCRIPTION
**Pull request by noituri**
Closes #798

### CODE REVIEW

File: scripts/verify_semver.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
```
Comment (wkozyra95) [inline scripts/verify_semver.sh:4]: do not validate semver, if anything then maybe verify if that image with this tag were already pulbished

File: .github/workflows/package_for_release.yml [hunk 2]
```diff
@@ -127,3 +144,23 @@ jobs:
...
+      steps:
+          - name: 📥 Checkout repo
+            uses: actions/checkout@v4
+
+          - name: 🔑 Log in to GitHub Container Registry
+            run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+          - name: 🔨 Build images
+            run: |
+              docker build -t ghcr.io/software-mansion/live_compositor:${{ inputs.version }} -f build_tools/docker/slim.Dockerfile .
```
Comment (wkozyra95) [inline .github/workflows/package_for_release.yml:159]:
```suggestion
              docker build -t ghcr.io/software-mansion/live-compositor:${{ inputs.version }} -f build_tools/docker/slim.Dockerfile .
```

File: .github/workflows/package_for_release.yml [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
 jobs:
+    verify-inputs:
+        runs-on: ubuntu-latest
+        steps:
+            - name: 📥 Checkout repo
+              uses: actions/checkout@v4
+
+            - name: 🔍 Verify inputs
+              run: ./scripts/verify_semver.sh ${{ inputs.version }}
     linux:
```
Comment (wkozyra95) [inline .github/workflows/package_for_release.yml:19]: Can we run only docker publish without building all the artifacts?

File: .github/workflows/package_for_release.yml [hunk 4]
```diff
@@ -127,3 +132,36 @@ jobs:
...
+          - name: 📤 Upload Live Compositor artifact
+            uses: actions/upload-artifact@v4
+            with:
+              name: docker_image_live_compositor.tar
+              path: docker_image_live_compositor.tar
+
+          - name: 🔨 Build Live Compositor with Web Renderer image
+            run: |
+              rm docker_image_live_compositor.tar
+              docker build -t ghcr.io/software-mansion/live-compositor-with-web-renderer:${{ inputs.version }} -f build_tools/docker/full.Dockerfile .
```
Comment (wkozyra95) [inline .github/workflows/package_for_release.yml:159]: all images should be `ghcr.io/software-mansion/live-compositor` use tags to mark we renderer variant

File: .github/workflows/package_for_release.yml [hunk 5]
```diff
@@ -127,3 +132,36 @@ jobs:
...
+      steps:
+          - name: 📥 Checkout repo
+            uses: actions/checkout@v4
+
+          - name: 🔑 Log in to GitHub Container Registry
+            run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+          - name: 🔨 Build Live Compositor image
+            run: |
+              docker build -t ghcr.io/software-mansion/live-compositor:${{ inputs.version }} -f build_tools/docker/slim.Dockerfile .
```
Comment (wkozyra95) [inline .github/workflows/package_for_release.yml:146]: can you run this image on macOS or do we need a separate arm version?

Comment (wkozyra95) [inline .github/workflows/package_for_release.yml:146]: can you run this image on macOS or do we need a separate arm version?

Comment (noituri) [reply .github/workflows/package_for_release.yml:146]: It works via x64 emulation

Comment (noituri) [reply .github/workflows/package_for_release.yml:146]: I tried running web renderer variant and on x64 it just hangs and doesn't start at all. On arm64 it works

File: scripts/release.sh [hunk 6]
```diff
@@ -39,4 +39,15 @@ gh release upload "$RELEASE_TAG" live_compositor_with_web_renderer_linux_x86_64.
 gh release upload "$RELEASE_TAG" live_compositor_with_web_renderer_darwin_x86_64.tar.gz
 gh release upload "$RELEASE_TAG" live_compositor_with_web_renderer_darwin_aarch64.tar.gz
 
+IMAGE_NAME="ghcr.io/software-mansion/live-compositor"
+COMMIT_HASH="$( git rev-parse HEAD )"
```
Comment (wkozyra95) [inline scripts/release.sh:43]: it needs to be an env here, this script will probably be called on a different commit

File: scripts/release.sh [hunk 7]
```diff
@@ -39,4 +39,15 @@ gh release upload "$RELEASE_TAG" live_compositor_with_web_renderer_linux_x86_64.
...
+docker pull "${IMAGE_NAME}:${COMMIT_HASH}"
```
Comment (wkozyra95) [inline scripts/release.sh:44]: move docker pull and stuff that can fail before release is created on github

File: .github/workflows/package_for_release.yml [hunk 8]
```diff
@@ -1,129 +1,171 @@
...
+    #             name: live_compositor_darwin_aarch64.tar.gz
+    #             path: live_compositor_darwin_aarch64.tar.gz
+    #
+    #         - uses: actions/upload-artifact@v4
+    #           with:
+    #             name: live_compositor_with_web_renderer_darwin_aarch64.tar.gz
+    #             path: live_compositor_with_web_renderer_darwin_aarch64.tar.gz
+    docker:
+      runs-on: ubuntu-latest
+      steps:
```
Comment (wkozyra95) [inline .github/workflows/package_for_release.yml:137]: split into separate jobs

File: .github/workflows/package_for_release.yml [hunk 9]
```diff
@@ -1,6 +1,11 @@
 name: package for release
 on:
-    workflow_dispatch: {}
+    workflow_dispatch:
+        inputs:
+            version:
+                type: string
+                description: Release version
+                required: true
```
Comment (wkozyra95) [inline .github/workflows/package_for_release.yml:8]:
```suggestion
    workflow_dispatch:
        inputs:
            docker_tag:
                type: string
                description: Docker image tag prefix
                required: false
```
skip docker build if not provided

**Review by wkozyra95 — COMMENTED** [review]
`live-compositor` -> `smelter`

**Review by wkozyra95 — APPROVED** [review]
